tests/testhpservice.nim

```diff
@@ -81,10 +80,13 @@ suite "Hole Punching":
     let hpservice = HPService.new(autonatService, autoRelayService, isPublicAddrIPAddrMock)
 
     let privatePeerSwitch = createSwitch(relayClient, hpservice)
+    let peerSwitch = createSwitch()
     let switchRelay = createSwitch(Relay.new())
 
-    await allFutures(switchRelay.start(), privatePeerSwitch.start(), publicPeerSwitch.start())
+    await allFutures(switchRelay.start(), privatePeerSwitch.start(), publicPeerSwitch.start(), peerSwitch.start())
 
+    await privatePeerSwitch.connect(peerSwitch.peerInfo.peerId, peerSwitch.peerInfo.addrs) # for autonat
+    await sleepAsync(200.millis) # wait for autorelay to kick in
```

Comment:
Try replacing with checkExpiring
And check if its a bug with auto relay @lchenut

Reply:
Same issue as before in Linux. Maybe my theory was wrong.